eucadeploy/plugins/debugger/debuggerplugin.py

```diff
@@ -25,7 +34,14 @@ def __init__(self, component_deployer):
         print cyan(self.message_style.format('DEBUG STARTING', self.name))
 
     def __del__(self):
-        self.report()
+        try:
+            disconnect_all()
+        except:
+            message = "Failed to disconnect from all hosts"
+            print red(self.message_style.format('DISCONNECT FAILED', message))
+            raise
```

Comment:
Do we ever hit the finally block if you raise here? We can just do the print and not raise i think. Also we should make that except clause more specific to the exception we are trying to catch, otherwise things will get gobbled up unnecessarily.